### ISSUE TITLE
Go to definition -> no definition found

### ISSUE THREAD

**Issue by TypedSource** [felixfbecker/vscode-php-intellisense#225]
i installed php-intellisense in the morning, disabled `php.suggest.basic`, set `php.executablePath` and opened my php project folder (inserted into workspace).

if i want to go to a definition of a class or function or var i allways get a message that no definition is available.

[http://prntscr.com/hk8x14](url)

the project was written in intelliJ and all works fine there. a change to vscode without correct working php intellisense is senseless.

**Comment by MadHed** [comment]
Do you see any errors under "Help" > "Developer Tools"

**Comment by TypedSource** [comment]
No, there is no error in the console and no error in the network tab. looks all fine.

nothing is triggered on trying to jump to definition

**Comment by rquispeq** [comment]
I have the same problem too, those options only work if i want a definition in the current file

**Comment by TypedSource** [comment]
no sollution for it? i think vscode is going to be useless for me as php coder under windows now and i switch back to intelliJ or Atom

**Comment by pacobyte** [comment]
**Go to Definition** is the first feature of this extension that I tried. Instead of getting an error, the "progressbar" just keeps moving across the screen as if it's constantly trying.

Does **Go to Definition** work for anyone? Or is there something specifically wrong with our installations?

**Comment by iEns** [comment]
I have the same problem.

1. I can only go to a definition in the same file.
2. If I open the file containing the definition, I can now go to the definition.
3. If I close the file containing the definition again, I can still go to the definition.
4. After restart of VS Code I am back to step 1.

Any way of forcing a scan or indexing of the workspace? 

(I'm on MacOS 10.13.3)

**Comment by cha0s** [comment]
Same. **Go to definition** only works for definitions that exist in already-opened files. This makes closing and reopening a PHP project extremely painful.

**Comment by rquispeq** [comment]
Hi, I remember that i coudn't fix this problem, but then i found a extension called Codeiginiter Intelliphense or something, and it has those options too. Of course, it only works if you have the framework.  It could help somebody :)

**Comment by benyaminl** [comment]
Does anyone here have try to reset the php.exe? Sometimes the php.exe isn't working and need to be killed.

**Comment by minkmk94** [comment]
I have same problem, i need this feature like "Go to definition" of Sublime text 3

**Comment by wagnerjsilva** [comment]
I definitely need to be able to go to definition as a PHP developer.
Shame, I really liked vs code, but this is an essential feature.

**Comment by tposcic** [comment]
Same issue here. Quite a pain in the ass

**Comment by MadHed** [comment]
Can anybody give more information? It works fine for me.

Extension version? Operating system? PHP Version? How many source files?

**Comment by Valera1980** [comment]
![default](https://user-images.githubusercontent.com/12474311/40978720-a03c72cc-68dc-11e8-9ea6-96e024da2e7c.png)

win 7 sp1

**Comment by danielhuang-030** [comment]
not working for me too, my info:
- OS: WINDOWS 10 Home
- VS Code ver
![20180731vscodever](https://user-images.githubusercontent.com/17424415/43436390-eed1a726-94b6-11e8-8fc2-690e80bf9089.jpg)
- VS Code ext
![20180731activeext](https://user-images.githubusercontent.com/17424415/43436411-ff00a188-94b6-11e8-8848-3152bd0f3093.jpg)
- VS Code user settings
![20180731usersettings](https://user-images.githubusercontent.com/17424415/43436431-0cef4f42-94b7-11e8-8bbe-0e86098096d3.jpg)
- function undefined
![20180731cannotfind](https://user-images.githubusercontent.com/17424415/43436461-3145ad46-94b7-11e8-950c-8dd0ba4ebd8a.jpg)

**Comment by benyaminl** [comment]
@danielhuang-030 When you tried the autocomplete does the autocomplete work? If not than don't expect that the goto function would work. I suggest to try https://marketplace.visualstudio.com/items?itemName=bmewburn.vscode-intelephense-client *I don't advertise it, but for now that's a alternative that works very good and not depends on php.exe*

**Comment by danielhuang-030** [comment]
@benyaminl Thank you for the suggestion, I just installed, it looks great!

**Comment by benyaminl** [comment]
@vasilestefirta change to Intelephense, it's more stable and more powerful than Intelisense https://marketplace.visualstudio.com/items?itemName=bmewburn.vscode-intelephense-client

**Comment by felixfbecker** [comment]
@benyaminl you already mentioned this before in the thread. Please keep this issue on topic.

To everyone else: If there is no update posted on this issue, there is no update. I don't have the capacity to investigate this in my free time. If anyone wants to do a PR with a fix, I am willing to invest time to review it. Given the amount of people here that posted here that they are affected by this, one would think someone here would care enough to look into it. It's just PHP code 🙂

**Comment by benyaminl** [comment]
@felixfbecker I think it's time for you to set up patreon profile so it would be your side project that generate money :v:

**Comment by vasilestefirta** [comment]
@felixfbecker Intelephense works great for me! Thanks for the suggestion 💯 👍

**Comment by Koekoe** [comment]
On MacOS, when a folder is a symlink to another folder the content is not indexed, thus some definitions are not working.  Could this be the problem ?

**Comment by cha0s** [comment]
The problem is that by default this extension doesn't scan all files. You have to do hax like patching the source, *or* making sure the first directory in your workspace is an aggregate of any other directories in your workspace (defeating the purpose of having them broken down cleanly in the first place).

The solution is to make the scan directories configurable, but I haven't written a patch for it yet.

**Comment by maslade** [comment]
@cha0s This was it for me, and I'm curious how many others in this thread are in similar situations. I had two folders: a project folder (mostly text files and assets) and a library folder (PHP).  "Go to Definition" only worked when the file containing the definition was already opened.

I then removed the project folder, so I only had the PHP folder, and now can navigate to those definitions even when the file is closed.

And for anyone wondering - I added the project folder back so that it was *second* on the list, and Intellisense still worked happily after a fresh restart. So it appears from behavior like the defect here is that the plugin only looks at the first directory in the list.

**Comment by moonbas3** [comment]
Works for me as well if the first folder in the project is the one including all the other code. Thanks for the tip.

**Comment by vushe** [comment]
> Works for me as well if the first folder in the project is the one including all the other code. Thanks for the tip.

Nope. Just endless scanning in my case... *Find all references* does not work too.

**Comment by ThomasK0lasa** [comment]
Same for me - the go to definition doesn't work in case of Joomla larg projects.

**Comment by Bashitha** [comment]
Same for me. Go to definition doesn't seem to be working specially when calling functions. I am using PHP in Visual Studio Code. OS is ubuntu. Auto complete is working fine. I installed PHP intelephense 0.8.8 extension. But still not working. Any updates on this issue?

**Comment by benyaminl** [comment]
> Same for me. Go to definition doesn't seem to be working specially when calling functions. I am using PHP in Visual Studio Code. OS is ubuntu. Auto complete is working fine. I installed PHP intelephense 0.8.8 extension. But still not working. Any updates on this issue?

Hello please it's Intelisense project repository. Intelephense Project repository is on https://github.com/bmewburn/intelephense/issues

**Comment by Bashitha** [comment]
> > Same for me. Go to definition doesn't seem to be working specially when calling functions. I am using PHP in Visual Studio Code. OS is ubuntu. Auto complete is working fine. I installed PHP intelephense 0.8.8 extension. But still not working. Any updates on this issue?
> 
> Hello please it's Intelisense project repository. Intelephense Project repository is on https://github.com/bmewburn/intelephense/issues

Hello @benyaminl Tried with Intellisense plugin also. But not functioning the go to definition for function calls. Just showed as loading without directing to the definition location.

**Comment by benyaminl** [comment]
> Same for me. Go to definition doesn't seem to be working specially when calling functions. I am using PHP in Visual Studio Code. OS is ubuntu. Auto complete is working fine. I installed PHP intelephense 0.8.8 extension. But still not working. Any updates on this issue?
> 
> Hello please it's Intelisense project repository. Intelephense Project repository is on https://github.com/bmewburn/intelephense/issues
> 
> Hello @benyaminl Tried with Intellisense extension also. But not functioning the go to definition for function calls. Just showed as loading without directing to the definition location.

If it's joomla seems there's some problem that you should know. Joomla code sometimes not follow the PSR standard and it's sequential in some way, that need special plugin for it. I also experience it with Laravel where the routing and other object not working and their code some follow the rule some not, Also Dolibarr ERP(based on PHP). You can't ask the universal tools to support code that's aren't standard. Sorry.

**Comment by ThomasK0lasa** [comment]
@benyaminl thank You for that information. :)
I am thinking about moving to PHP Storm because everything works fine in it. But I like the VSC. :(
I have same problem with Intelephense and Intelisense in case of Joomla and VSC. I am still thinking that maybe I have some missconfiguration.

**Comment by benyaminl** [comment]
> @benyaminl thank You for that information. :)
> I am thinking about moving to PHP Storm because everything works fine in it. But I like the VSC. :(
> I have same problem with Intelephense and Intelisense in case of Joomla and VSC. I am still thinking that maybe I have some missconfiguration.

Not everything working in PHP Strom, Dolibarr isn't supported on both of editor, and only class, variable and such that's follow the PSR standard that's indexed. So for me it's normal for VS Code. PHP Strom is too heavy and not power friendly

**Comment by vushe** [comment]
> > Same for me. Go to definition doesn't seem to be working specially when calling functions. I am using PHP in Visual Studio Code. OS is ubuntu. Auto complete is working fine. I installed PHP intelephense 0.8.8 extension. But still not working. Any updates on this issue?
> 
> Hello please it's Intelisense project repository. Intelephense Project repository is on https://github.com/bmewburn/intelephense/issues

You've mentioned it three times already, whereas one time would be more than enough. Do you want to start discussing here bugs of PHP Intelliphense too? Politeness is highly appreciated.

**Comment by benyaminl** [comment]
> Same for me. Go to definition doesn't seem to be working specially when calling functions. I am using PHP in Visual Studio Code. OS is ubuntu. Auto complete is working fine. I installed PHP intelephense 0.8.8 extension. But still not working. Any updates on this issue?
> 
> Hello please it's Intelisense project repository. Intelephense Project repository is on https://github.com/bmewburn/intelephense/issues
> 
> You've mentioned it three times already, whereas one time would be more than enough. Do you want to start discussing here bugs of PHP Intelliphense too? Politeness is highly appreciated.

I don't talk about intelephense bug, but many people refer it too much. I think I just refer them to it and stop talking OOT, sorry if it's seems so pushy, but it's intelisense project and not for other extension mater. That's why I remind them.

**Comment by dgloriaweb** [comment]
Hi, I use vscode with a codeigniter php project, and I'm making a simple model-controller stuff using CI as base, but the $this->post_model->get_posts(); isn't picking up the get_posts from the method. I also work on a huge project at work, where this would be essential, and it's quite frustrating, because I used to use phpstorm and it worked like charm.

**Comment by nicolaib** [comment]
I can't tell why it worked, but I had this problem too and went to settings.json and changed the path to php.exe to have double backslashes instead of slashes. I restarted  VSCode and therefore also closed another VSCode instance window. After this restart the problem was gone. Maybe it will help someone. 
I tried to recreate the problem afterwards and couldn't (even with changing back to slashes), so there might be a caching/loading issue going on.

**Comment by karensrapyan** [comment]
I see that in https://intelephense.com/ they are selling that option.
Open source VS Code:)

**Comment by yehudaTiram** [comment]
For me the issue was that vs code was using php v7.0.3
To fix it:
Go to ~\AppData\Roaming\Code\User\settings.json (Windows 10)
OR
Click F1 to open Command palette
search and select   "Preferences: Open settings (UI)
search for "php.validate.executablePath"
Click edit in settings.json 
in the json file change the path to the latest php version in your machin.

(for me it is "php.validate.executablePath": "C:/wamp64/bin/php/php7.4.0/php.exe",)

**Comment by szczemp** [comment]
I have similar problem. And with separate functions with the same name. Eg: myCustomFunction() in adminInclude.php file and myCustomFunction() in indexInclude.php. adminInclude.php file is included in admin.php and there I use myCustomFunction(). indexInclude.php is included in index.php and there I use myCustomFunction() too. Go To Definition shold find definions in both files, but there is only one sometimes from adminInclude.php someitmes from indexInclude.php.

**Comment by szczemp** [comment]
> I have similar problem. And with separate functions with the same name. Eg: myCustomFunction() in adminInclude.php file and myCustomFunction() in indexInclude.php. adminInclude.php file is included in admin.php and there I use myCustomFunction(). indexInclude.php is included in index.php and there I use myCustomFunction() too. Go To Definition shold find definions in both files, but there is only one sometimes from adminInclude.php someitmes from indexInclude.php.

I think I found a solution. I installed the extension: https://marketplace.visualstudio.com/items?itemName=DEVSENSE.phptools-vscode Go To Definition works fine.
